x/nft/approval_ops.go

```diff
@@ -0,0 +1,72 @@
+package nft
+
+import (
+	"errors"
+
+	"github.com/iov-one/weave"
+)
+
+type ApprovalOps struct {
+	owner weave.Address
+	//TODO: Possibly define a type for it, e.g. *ActionApprovalsSet
+	approvals *[]*ActionApprovals
+}
+
+//TODO: Sort errors and their codes
+func NewApprovalOps(nft Owned, approvals *[]*ActionApprovals) *ApprovalOps {
+	return &ApprovalOps{owner: nft.OwnerAddress(), approvals: approvals}
+}
+
+func (o *ApprovalOps) List() Approvals {
+	res := make(map[string]ApprovalMeta, 0)
+	for _, v := range *o.approvals {
+		res[v.Action] = v.Approvals
+	}
+	return res
+}
+
+func (o *ApprovalOps) Revoke(action string, from weave.Address) error {
+	if from == nil || from.Equals(o.owner) {
+		return errors.New("invalid account")
+	}
+	approvalsToRemove := o.List().ForAction(action).ForAddress(from)
+	if len(approvalsToRemove) == 0 {
+		return errors.New("does not exist")
+	}
+	for _, a := range approvalsToRemove {
+		for _, v := range a {
+			if v.Options.Immutable {
+				return errors.New("immutable and can not be changed")
+			}
+		}
+	}
+	*o.approvals = o.List().Filter(approvalsToRemove).AsPersistable()
+	return nil
+}
+
+//TODO: Note, that we are always excluding an owner from permissions, that might not be ideal
```

Comment:
owner implicitly has rights to do everything on related to the token (approve, transfer, etc)

Reply:
Yes, so we are correctly not adding him to approvals